src/utils/useUpdateEffect/useUpdateEffect.ts

```diff
@@ -0,0 +1,43 @@
+import * as React from 'react';
+
+// useUpdateEffect is a useEffect that doesn't fire on initial render
+// for detecting when values change
+// https://stackoverflow.com/a/58217148
+
+const useIsMounted = function useIsMounted() {
```

Comment:
I'm struggling to understand what's going on here, the assignment of a named function to the same-named variable. I'd think that the function declaration name would either be different from the variable it's assigned to so that it could use its own name for recursion, or... well I'm not sure what else for. :)

Is it for hoisting for some reason?

Reply:
My understanding is that this is a convention some people use because it is convenient for stack tracing. You name your function with the same name so it isn't merely anonymous.

Reply:
Ok, I can see that. Could you add a comment to that effect? I'm probably not the only person who will go, "Whaaaat?"